File: src/test/testdata/inspections/unsafeIterators/SortedIntList.java
package main;

import com.badlogic.gdx.utils.ObjectSet;
import com.badlogic.gdx.utils.SortedIntList;

class SortedIntListTest {

  void test() {
    SortedIntList<ObjectSet<String>> list = new SortedIntList<ObjectSet<String>>();

    for (SortedIntList.Node<ObjectSet<String>> n: <warning>list</warning>) {
      for (String s: <warning>n.value</warning>) {

      }
    }

    <warning><warning>list.iterator()</warning>.next().value.iterator()</warning>.next();
  }

}

File: src/test/testdata/inspections/unsafeIterators/SortedIntList.kt
package main

import com.badlogic.gdx.utils.ObjectSet
import com.badlogic.gdx.utils.SortedIntList

class SortedIntListTestKotlin {

  fun test() {
    val list = SortedIntList<ObjectSet<String>>()

    for (n in <warning>list</warning>) {
      for (s in <warning>n.value</warning>) {

    }
    }

    <warning><warning>list.iterator()</warning>.next().value.iterator()</warning>.next()

    list.let {
      <warning>it.iterator()</warning>
    }
  }

}
